### PR TITLE
feat!: require blank lines before and after line comments

### LINKED ISSUE #169: Enforce blank line before and after comments
This corresponds to https://github.com/liferay/liferay-npm-tools/issues/409

I originally created it over there because it was originally related just to SCSS, and liferay-npm-tools is where we have our stylelint tooling located. But the request got extended to cover JS as well, which is going to involve ESLint and therefore eslint-config-liferay as well, so I am creating this one over here to reflect that.

### PR DESCRIPTION
**Pull request by wincent**
This basically matches the liferay-portal style, which is:

-  Block comments: blank line before, no blank after.
-  Line comments: blank line before and after.

With these tweaks that are required for compatibility with Prettier:

-  No blank lines before comments at the start of a block, an object, an array, or a class.
-  No blank lines after comments at the end of a block, an object, an array, or a class.

And built-in exceptions provided by ESLint:

-   Block comments containing ESLint directives are not enforced; eg:

    ```
    /* eslint-disable-next-line no-console */
    console.log('no blank between directive comment and this call');
    ```

Docs: https://eslint.org/docs/rules/lines-around-comment

Closes: https://github.com/liferay/eslint-config-liferay/issues/169

### CODE REVIEW

**Comment by wincent** [comment]
Before deciding whether this is mergeable, want to test it out in liferay-portal and see if it wreaks havoc or not.

**Comment by wincent** [comment]
In liferay-portal:

```
cp ~/code/eslint-config-liferay/index.js modules/node_modules/eslint-config-liferay
portool yarn run liferay-npm-scripts lint
```

I get:

```
✖ 124 problems (124 errors, 0 warnings)
  124 errors and 0 warnings potentially fixable with the `--fix` option.
```

Autofix changes (52 files changed, 119 insertions(+)):

https://gist.github.com/wincent/188825d9713978ef498b790bc4797097

Output is *almost* stable: ie. if I run it a second time and get no changes, but If I run a "check" I get some files which report problems due to things like commented-out JSX properties, which should be removed anyway; eg. stuff like [this file in questions-web](https://github.com/liferay/liferay-portal/blob/957768512c5fa1b6d33724f33ebe177ceb81f408/modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/Rating.es.js#L75-L83):

```
			<ClayButton
				className={
					'text-reset' + (userRating === -1 ? ' text-primary' : '')
				}
				displayType="unstyled"
				// small={if-it-is-a-sub-comment}

				monospaced
				onClick={() => voteChange(-1)}
			>
```

ESLint wants a blank here before and after the comment, Prettier doesn't care about the line after the comment, but [doesn't want one to be before it](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdG70B4DCAbAQwE8AhAVxhmlSkw3rEIGcmA5AgWzgF5ga6BMAOTwAHjAC0AJzhM4wgAQBqBQAoycqQCUCMAJZQA5gu6mFEgIwKA-AqEKxkgA5S9HAlKL2kdoQEp+ejoAX0DBABM9JidCIgAVIiceVBAyKCYYIjw4cJSaBQLCoqKwoJgAenKFJnc8PF49ADMJPUkoiQIJJjIAIwlIDi5YUKhiscLSug5oCGiCMBzJ+mh8PTAAa15VPxMAPgUANwh4HAALAiM4VUs-EcFdrHLdkAAaEAgnfWgmZFAPKQgAHcAAoeBA-FAEPCA4g-N49KTzdbyADKTnmBkMyBgUjIcDepxgHDwAHVTq1ZOiFijwa09AdWkRkOAWK8QAZNDBgYjDO5kI0oXI3gArJiiEiIjaozhwAAyBjg-MF+JAotEKMx2QAimRjoqkAK8EKQOipJpmT0CD04Hg2S4DDASXpwjBTsgABwABjeLlmcBJiKczJcsjgUgOireAEddfBuR8ISACEwJFA4DkcmyZDG9DJuQReQQlUaVXIOHpsbjS5q4Dq9cXjTArU6XW6kAAmN44gh6PCYnAQQZFlCyACsbI0cDiVohhuNBzxAEkoOEEDAUWBXJ8AIIrlGZbINuDBYJAA).

Like I said, we should just remove those comments.

Pretty tame, but I guess I should have expected that, because liferay-portal has almost no comments.

File: plugins/eslint-plugin-liferay-portal/lib/rules/no-explicit-extend.js [hunk 1]
```diff
@@ -60,6 +63,7 @@ function fix(nodesToRemove, context, fixer) {
 
 	// Record index of last `item` in `items` that will remain after we've
 	// deleted `nodesToRemove`.
+
```
Comment (bryceosterhaus) [inline plugins/eslint-plugin-liferay-portal/lib/rules/no-explicit-extend.js:66]: looks so odd to have a line after in-line comments. Almost as if the code it was commenting was deleted. but then again, its all arbitrary 🤷‍♂️

**Comment by wincent** [comment]
Nothing else is in flight right now, so may as well proceed with getting this into liferay-portal.

**Comment by wincent** [comment]
Currently testing liferay-portal update PR [over here](https://github.com/wincent/liferay-portal/pull/241).

**Pull request merged**